chainindex/ddls.go

```diff
@@ -0,0 +1,55 @@
+package chainindex
```

Comment:
I can't say I love having a new top-level package, could we tuck this into chain/ perhaps?

Reply:
I've removed the old `chain/index` package completely and we now only have this one in place.

Reply:
still don't love a top level package, there's already too many IMO, `chain/index` seems like a good place for this to live even if it's a departure from what's there

Reply:
Fixed this by moving this `chain/index`.